build/xtask/src/dist.rs

```diff
@@ -318,6 +320,48 @@ pub fn package(
         let s =
             secure_update(&cfg, allocs, &mut all_output_sections, image_name)?;
 
+        // Add an empty output section for the caboose
+        //
+        // This has to be done before building the kernel, because the caboose
+        // is included in the total image size that's patched into the kernel
+        // header.
+        if let Some(caboose) = &cfg.toml.caboose {
+            if (caboose.size as usize) < std::mem::size_of::<u32>() * 2 {
+                bail!("caboose is too small; must fit at least 2x u32");
+            }
+
+            for t in &caboose.tasks {
+                if !cfg.toml.tasks.contains_key(t) {
+                    bail!("caboose specifies invalid task {t}");
+                }
+            }
+
+            let (_, caboose_range) = allocs.caboose.as_ref().unwrap();
+            // The caboose has the format
+            // [CABOOSE_MAGIC, ..., MAX_LENGTH]
+            // where all words in between are initialize to u32::MAX
```

Comment:
Yup, `u32::MAX` is because we have to pick _something_, and that's free on the SP after the section is erased.

It's worth noting that the caboose will _still_ be written even if no one has modified the archive to put data in there; it will just be all `0xFF`.  This means that on the LPC55, we won't be leaving this region in the troublesome erased-but-not-written state.